src/sql/workbench/contrib/editData/browser/editDataGridPanel.ts

```diff
@@ -497,8 +497,12 @@ export class EditDataGridPanel extends GridParentComponent {
 			this.firstLoad = false;
 		}
 		else {
-
-			this.table.setData(this.gridDataProvider);
+			try {
+				this.table.setData(this.gridDataProvider);
```

Comment:
@alanrenmsft he did this because the data provider class is in the base/ui folder which isn't allowed

So with that being said and in general this just being a "quick fix" then what you had before @smartguest seems fine. 

Although doesn't this indicate a problem with the stuff you added a while back @alanrenmsft ? https://github.com/microsoft/azuredatastudio/pull/14833 Do we just not allow filtering when the data is "in-memory"?

It'd be nice to fix this so it the async provider actually implements filtering though at some point so we don't risk this breaking something else down the line...